modules/videoio/src/cap_msmf.cpp

```diff
@@ -1318,6 +1320,19 @@ bool CvCapture_MSMF::setAudioProperties(const cv::VideoCaptureParameters& params
             outputAudioFormat = value;
         }
     }
+    if (params.has(CAP_PROP_AUDIO_SAMPLES_PER_SECOND))
+    {
+        int value = static_cast<int>(params.get<double>(CAP_PROP_AUDIO_SAMPLES_PER_SECOND));
+        if (value != 8000 && value != 11025 && value !=  22050 && value !=  44100)
+        {
+            CV_LOG_ERROR(NULL, "VIDEOIO/MSMF: CAP_PROP_AUDIO_SAMPLES_PER_SECOND parameter value is invalid/unsupported: " << value);
+            return false;
+        }
```

Comment:
I believe MSMF framework should know better which data rate is supported. There is no limitation on OpenCV side (except may be >= 0 validation check). No need to duplicate MSMF logic in OpenCV code - no way to maintain this approach.
Need to ensure that MSMF should properly return "false" from `.open()` with unsupported parameter value.